waiter/src/waiter/state.clj

```diff
@@ -1256,7 +1271,11 @@
                                        deployment-error (get-deployment-error healthy-instances unhealthy-instances failed-instances deployment-error-config)
                                        service-id->deployment-error' (if deployment-error
                                                                        (assoc service-id->deployment-error service-id deployment-error)
-                                                                       (dissoc service-id->deployment-error service-id))]
+                                                                       (dissoc service-id->deployment-error service-id))
+                                       oom-instability (get-instability-issue failed-instances)
+                                       service-id->oom-instability' (if oom-instability
```

Comment:
I think we should include the `oom-instability` in the `if` clause and maintain a set for OOM instability instead of `assoc`-ing `true` into a map. The new field could be called `oom-instable-service-ids`